src/algo/camera-age/l500/camera-age.cpp

```diff
@@ -44,15 +44,24 @@ unsigned work_week::operator-( const work_week & ww ) const
 // XXXX � Sequential number
 work_week get_manufature_work_week( const std::string & serial )
 {
+    if( serial.size() != 8 )
+        throw invalid_value_exception( "invalid serial number: " + serial + " is of invalid size" );
     unsigned Y = serial[1] - '0';  // Converts char to int, '0'-> 0, '1'-> 1, ...
     unsigned man_year = 0;
     // using Y from serial number to get manufactoring year
     if( Y == 9 )
         man_year = 2019;
-    else
+    else if( Y >= 0 && Y <= 8 )
```

Comment:
If Y is `unsigned`, it will always be 0 or more. Need to check that the original digit is between `0` and `9`